adapters/airflow_connector.py

```diff
@@ -24,34 +35,45 @@ def query(self, request):
     Returns:
         The database result (tuple)
     """
-    cursor = self._airflow_db.cursor()
     # The following ensures that the query executes and returns,
     # even if the db connection has been lost
+    logging.info('Running query "%s" against Airflow DB' % request)
     try:
+      cursor = self._airflow_db.cursor()
       cursor.execute(request)
       response = cursor.fetchall()
     except MySQLdb.OperationalError, e:
-      if e[0] == 2006:
-        self._airflow_db = self.create_connection()
+      logging.error('Error %i triggered with MySQLdb: %s'  % (e[0], e[1]))
+      if e[0] in [errorcode.CR_SERVER_GONE_ERROR, errorcode.CR_SERVER_LOST]:
+        if cursor:
+          cursor.close()
+        self._airflow_db = self._create_connection()
         cursor = self._airflow_db.cursor()
         cursor.execute(request)
         response = cursor.fetchall()
+        logging.info('MySQLdb connection restored.')
 
     cursor.close()
 
     return response
 
-  def create_connection(self):
+  def _create_connection(self):
```

Comment:
Do you have to return the db ? or can you just update self._airflow_db directly ?

Reply:
Now updates variable directly